enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,138 @@
+* The UI will clearly indicate when this upgrade blocking mechanism is blocking upgrades, why, and what needs to be done to clear the block.
+* This upgrade blocking mechanism must be compatible with SD, ACM, IKS/ROKS, Hive, and Hypershift.
+
+### Non-Goals
+
+## Proposal
+
+CVO's existing framework for blocking minor level upgrades will be leveraged.
+The existing framework periodically checks upgradeable conditions.
+A new [upgradeable check method][upgradeableCheck-interface] will be added to the existing framework to implement this upgrade blocking mechanism.
+Use of this existing framework also allows this new upgrade blocking mechanism to be bypassed by using the `force` upgrade option.
+
+The new method will check a newly defined configmap for one or more predefined keys.
+Each key represents a specific upgrade blocking condition that must be acknowledged before minor level upgrades are allowed.
+An administrator acknowledges a condition by adding the corresponding key with a value of `true` to the configmap.
+Values other than `true` will be ignored and therefore not clear the upgrade block.
+The keys will be named to make it explicitly clear what the administrator is acknowledging and thereby agreeing to.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to the next minor release until it finds each of the predefined keys in the configmap with a value of `true`.
+For each key blocking an upgrade, CVO will generate a detailed message pointing to relevant OCP documentation explaining what the administrator is expected to do before acknowledging task completion, i.e. entering the corresponding key into the configmap with a value of `true`.
+
+The new configmap `upgrade-ack` will be installed as part of the release in the namespace `openshift-cluster-version`.
+It will initially have no keys since the administrator is expected to add the relevant key as part of acknowledgement.
+
+### User Stories
+
+Clusters upgraded to 4.9 will no longer have access to these [APIs][k8s-1.22-removed-apis].
+This new upgrade blocking mechanism will be used in an attempt to require administrators to read detailed documentation regarding the removed APIs and the possible impact.
+This OCP documentation will list which APIs have been removed and how an administrator checks their cluster for use of those APIs.
+
+4.8.z versions that are upgradeable to 4.9 will contain this enhancement and have the configmap `openshift-cluster-version/upgrade-ack`.
+CVO will set `Upgradeable=false` which in turn blocks cluster upgrades to 4.9 until the administrator has added the key/value pair `ack-4.9-api-removal-upgrade-approved: "true"` to the configmap.
+Until this is done the `Conditions` console page will display an `Upgradeable` condition message directing the administrator to the relevant OCP documentation.
+
+If the administrator wants to upgrade to 4.9 without performing the acknowledgment `force` can be used.
+
```

Comment:
user stories are generally more in the form of:

as a cluster admin, i want to be warned that apis will be removed if i upgrade to 4.9 before i accidentally upgraded and be instructed how to check my cluster for potential impact

as a cluster admin i want to be able to override the ack requirement and upgrade anyway

as an OCP developer i want a way to introduce additional ACKs that admins will be required to set before upgrading to future minor OCP versions

as an ocp product team we do not want customers to be surprised by workloads breaking due to apis being removed after they upgrade


etc

Reply:
Agree, not user stories. I put it under a new heading. I would have left the `User Stores` section empty since it seems requirements are pretty well flushed out but since you already had some good ones I put them in and will think about it some more.